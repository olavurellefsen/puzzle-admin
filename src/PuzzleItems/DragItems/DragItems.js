import React, { useContext, Fragment } from "react";
import { MdAdd } from "react-icons/md";
//import LineTo from "../../Utils/lineto";
import MainContext from "../../Context";
import {
  DragItemsContainer,
  DragItemsHeader,
  DragItemsBox,
  DragPlaceholder,
  AddDragItem
} from "./DragItems.style";
import DragItem from "./DragItem/DragItem";
import { Subscription, Mutation } from "react-apollo";
import gql from "graphql-tag";

export default () => {
  const [state] = useContext(MainContext);

  let currentPuzzle = state.currentPuzzles.find(
    puzzle => puzzle.scene_id === state.currentScene
  );

  const INSERT_DRAGITEM = gql`
    mutation insert_dragitem($puzzle_id: Int!) {
      insert_dragitem(
        objects: [
          {
            puzzle_id: $puzzle_id
            wait: false
            puzzleItemBypuzzleItemId: {
              data: {
                imagefile: "placeholder.png"
                name: "new drag item"
              }
            }
          }
        ]
      ) {
        returning {
          id
        }
      }
    }
  `;

  const SUBSCRIBE_DRAGITEM = gql`
    subscription get_dragitem($puzzle_id: Int!) {
      dragitem(
        order_by: { sequence: asc }
        where: {
          _and: [
            { puzzle_id: { _eq: $puzzle_id } }
            {
              puzzleItemBypuzzleItemId: {
                itemBypuzzletextKey: { language_id: { _eq: 1 } }
              }
            }
          ]
        }
      ) {
        id
        puzzle_id
        sequence
        wait
        puzzleItemBypuzzleItemId {
          id
          imagefile
          name
          puzzletext_key
          itemBypuzzletextKey {
            value
          }
        }
        draglogicsBydragitemId {
          targetitem_id
        }
      }
    }
  `;

  if (typeof currentPuzzle === "undefined") {
    return "Loading puzzle...";
  } else {
    let currentPuzzleId = currentPuzzle.puzzle_id;

    const dragItems = (
      <Subscription
        subscription={SUBSCRIBE_DRAGITEM}
        variables={{ puzzle_id: currentPuzzleId }}
      >
        {({ data, loading }) => {
          if (loading) {
            return (
              <>
                <DragPlaceholder className="drag1">Loading...</DragPlaceholder>
                <DragPlaceholder className="drag2" />
                <DragPlaceholder className="drag3" />
                <DragPlaceholder className="drag4" />
              </>
            );
          } else {
            const dragitems = data.dragitem;
            return dragitems.map((dragitem, index) => {
              /* const lines = dragitem.draglogicsBydragitemId.map(
                (logicitem, index2) => {
                  return (
                    <LineTo
                      key={index2}
                      from={`drag${dragitem.id}`}
                      to={`target${logicitem.targetitem_id}`}
                      fromAnchor="middle right"
                      toAnchor="middle left"
                      borderWidth={3}
                      delay={1000}
                      zIndex={999}
                    />
                  );
                }
              );*/
              return (
                <Fragment key={index}>
                  <DragItem data-testid="DragItem" key={index} index={index} dragitem={dragitem} />
                </Fragment>
              );
            });
          }
        }}
      </Subscription>
    );

    const insertDragItem = (
      <Mutation mutation={INSERT_DRAGITEM}>
        {(insertDragItem, { data }) => (
          <AddDragItem>
            <MdAdd
              onClick={() => {
                insertDragItem({
                  variables: { puzzle_id: currentPuzzleId }
                });
              }}
            />
          </AddDragItem>
        )}
      </Mutation>
    );

    return (
      <DragItemsContainer data-testid="DragItemsContainer">
        <DragItemsHeader>H√ÅLIMYNDIR</DragItemsHeader>
        <DragItemsBox>
          {dragItems}
        </DragItemsBox>
      </DragItemsContainer>
    );
  }
};
