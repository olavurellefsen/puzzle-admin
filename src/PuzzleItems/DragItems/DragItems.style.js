import styled from 'styled-components';

export const DragItemsContainer = styled.div`
  min-width: 294px;
  overflow: hidden;
`;

export const DragItemsHeader = styled.div`
  text-align: left;
  padding-top: 10px;
  padding-left: 20px;
  font-size: 16px;
  color: #666666;
`;

export const DragItemsBox = styled.div`
  width: 100%;
  height: 100%;
  overflow: auto;
  padding-right: 14px;
`;

export const DragPlaceholder = styled.div`
  height: 200px;
  width: 200px;
  margin: 20px 20px;
`;

export const AddDragItem = styled.div`
  height: 200px;
  width: 200px;
  margin: 20px 20px;
  font-size: 32px;
  text-align: center;
  cursor: pointer;
`;