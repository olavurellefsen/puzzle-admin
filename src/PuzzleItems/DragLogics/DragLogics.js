import React from "react";
import { DragLogicsContainer, DragLogicsImage } from "./DragLogics.style";

export default () => (
  <DragLogicsContainer data-testid="DragLogicsContainer">
    <DragLogicsImage
      src={"https://via.placeholder.com/200x800?text=Logic"}
      data-testid="DragLogicsImage"
    />
  </DragLogicsContainer>
);
