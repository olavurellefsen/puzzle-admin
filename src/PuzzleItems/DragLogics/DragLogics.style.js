import styled from 'styled-components';

export const DragLogicsContainer = styled.div`
`;