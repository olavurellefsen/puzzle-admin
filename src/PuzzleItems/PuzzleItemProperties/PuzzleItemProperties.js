import React from 'react';
import { PuzzleItemPropertiesContainer } from './PuzzleItemProperties.style';

export default () => <PuzzleItemPropertiesContainer data-testid="PuzzleItemPropertiesContainer"/>;