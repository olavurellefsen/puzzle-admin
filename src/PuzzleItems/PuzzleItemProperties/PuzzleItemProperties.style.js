import styled from 'styled-components';

export const PuzzleItemPropertiesContainer = styled.div`
  margin: 10px 10px;
`;

export const PuzzleItemPropertiesImage = styled.img`
`;
