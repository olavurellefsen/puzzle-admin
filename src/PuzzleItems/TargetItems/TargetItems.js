import React, { useContext } from "react";
import MainContext from "../../Context";
import { TargetItemsContainer, TargetPlaceholder } from "./TargetItems.style";
import TargetItem from "./TargetItem/TargetItem";
import { Subscription } from "react-apollo";
import gql from "graphql-tag";

export default () => {
  const [state] = useContext(MainContext);

  let currentPuzzle = state.currentPuzzles.find(
    puzzle => puzzle.scene_id === state.currentScene
  );
  if (typeof currentPuzzle === "undefined") {
    return "Loading puzzle...";
  } else {
    let currentPuzzleId = currentPuzzle.puzzle_id;
    const subscription = gql`
    subscription {
      targetitem(order_by: { sequence: asc }, where: { puzzle_id: { _eq: ${currentPuzzleId} } }) {
        id
        puzzle_id
        sequence
        puzzleItemBypuzzleItemId {
          imagefile
          name
        }
      }
    }
    `;

    return (
      <TargetItemsContainer data-testid="TargetItemsContainer">
        <Subscription subscription={subscription}>
          {({ data, loading }) => {
            if (loading) {
              return (
                <>
                  <TargetPlaceholder className="target1">
                    Loading...
                  </TargetPlaceholder>
                  <TargetPlaceholder className="target2" />
                  <TargetPlaceholder className="target3" />
                  <TargetPlaceholder className="target4" />
                </>
              );
            } else {
              let Targetitems = data.targetitem;
              return Targetitems.map((targetitem, index) => {
                return (
                  <TargetItem
                    key={index}
                    index={index}
                    targetitem={targetitem}
                  />
                );
              });
            }
          }}
        </Subscription>
      </TargetItemsContainer>
    );
  }
};
