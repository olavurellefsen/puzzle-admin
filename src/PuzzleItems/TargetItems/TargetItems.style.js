import styled from 'styled-components';

export const TargetItemsContainer = styled.div`
`;
TargetItemsContainer.displayName = 'TargetItemsContainer';