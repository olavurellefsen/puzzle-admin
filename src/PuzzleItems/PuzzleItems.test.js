import React from 'react';
import { shallow } from 'enzyme';
import PuzzleList from './PuzzleItems';
import DragItems from './DragItems/DragItems';
import DragLogics from './DragLogics/DragLogics';
import TargetItems from './TargetItems/TargetItems';
import PuzzleItemProperties from './PuzzleItemProperties/PuzzleItemProperties';

describe('PuzzleList', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<PuzzleList />));
  it('should render correctly', () => expect(wrapper).toMatchSnapshot());
  it('should render a PuzzleItemsContainer', () => {
    expect(wrapper.find('PuzzleItemsContainer').length).toEqual(1);
  });
  it("should render the DragItems component", () => {
    expect(wrapper.containsMatchingElement(<DragItems />)).toEqual(true);
  });
  it("should render the DragLogics component", () => {
    expect(wrapper.containsMatchingElement(<DragLogics />)).toEqual(true);
  });
  it("should render the TargetItems component", () => {
    expect(wrapper.containsMatchingElement(<TargetItems />)).toEqual(true);
  });
  it("should render the PuzzleItemProperties component", () => {
    expect(wrapper.containsMatchingElement(<PuzzleItemProperties />)).toEqual(true);
  });

});