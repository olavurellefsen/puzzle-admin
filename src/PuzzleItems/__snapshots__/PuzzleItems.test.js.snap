// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa lhXlbL"
    data-testid="PuzzleItemsContainer"
  >
    <div
      class="sc-bwzfXH htoLMg"
      data-testid="PuzzleItemsBox"
    >
      <div
        class="sc-htpNat fdfuso"
        data-testid="DragItemsContainer"
      >
        <div
          class="drag1 sc-bxivhb kHGHfL"
        >
          Loading...
        </div>
        <div
          class="drag2 sc-bxivhb kHGHfL"
        />
        <div
          class="drag3 sc-bxivhb kHGHfL"
        />
        <div
          class="drag4 sc-bxivhb kHGHfL"
        />
      </div>
      <div
        class="sc-gZMcBi ggQDYT"
        data-testid="TargetItemsContainer"
      >
        <div
          class="target1 sc-gqjmRU hygCTt"
        >
          Loading...
        </div>
        <div
          class="target2 sc-gqjmRU hygCTt"
        />
        <div
          class="target3 sc-gqjmRU hygCTt"
        />
        <div
          class="target4 sc-gqjmRU hygCTt"
        />
      </div>
    </div>
    <div
      class="sc-kgoBCf INKp"
      data-testid="PuzzleItemPropertiesContainer"
    >
      <img
        class="sc-kGXeez fqVwMj"
        data-testid="PuzzleItemPropertiesImage"
        src="https://via.placeholder.com/640x200?text=PuzzleItemProperties"
      />
    </div>
  </div>
</div>
`;
